include/flatbuffers/flatbuffers.h

```diff
@@ -1273,8 +1302,12 @@ class FlatBufferBuilder {
   void PopBytes(size_t amount) { buf_.pop(amount); }
 
   template<typename T> void AssertScalarT() {
+    // clang-format off
+    #ifndef FLATBUFFERS_NO_STL
     // The code assumes power of 2 sizes and endian-swap-ability.
     static_assert(flatbuffers::is_scalar<T>::value, "T must be a scalar type");
+    #endif  // FLATBUFFERS_NO_STL
```

Comment:
How `is_scalar` or other type traits in code connected with NO_STL (std::vector and std::string)?